tests/unit/helpers.js

```diff
@@ -158,22 +158,15 @@ export function makeSureItFails(): Function {
 export function fake<T>(
   original: Object, methods: Object = {}, skipProperties: Array<string> = []
 ): T {
-  var stub = {};
+  const stub = {};
 
   // Provide stubs for all original members:
-  var props = [];
-  var obj = original;
-  while (obj) {
-    props = props.concat(Object.getOwnPropertyNames(obj));
-    obj = Object.getPrototypeOf(obj);
-  }
+  const proto = Object.getPrototypeOf(original);
+  const props = Object.getOwnPropertyNames(original).concat(
+    proto ? Object.getOwnPropertyNames(proto) : []
```

Comment:
According to the current implementation, `proto` is expected to be non-void.
It's not necessary to check for emptiness of `proto`. If you do, and want to be resilient against arbitrary values, then for completeness you should also add `proto && ` to the loop below, in case the property has a falsey value (e.g. `{__proto__: nul, propname: false}`).

Reply:
Thanks for pointing it out. 

I opted to simplify it by considering the `proto` to always be non-void (we don't have any such case, better to keep it simple).